apps/src/storage/levelbuilder/Dataset.jsx

```diff
@@ -14,17 +37,58 @@ class Dataset extends React.Component {
     // it as an array."
     // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
     tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
-      .isRequired
+      .isRequired,
+    onUploadComplete: PropTypes.func.isRequired
+  };
+
+  state = {
+    notice: null,
+    isError: false
+  };
+
+  displayNotice = (notice, isError) => {
+    this.setState({notice, isError}, () =>
+      setTimeout(() => this.setState({notice: null, isError: false}), 5000)
```

Comment:
For the success state, clearing the message makes sense, but for the error state, would we want to keep the message there?